plugins/processors/ec2tagger/ec2tagger.go

```diff
@@ -309,24 +243,23 @@ func (t *Tagger) Init() error {
 			t.metadataLookup.instanceId = true
 		case mdKeyImageId:
 			t.metadataLookup.imageId = true
-		case mdKeyInstaneType:
+		case mdKeyInstanceType:
 			t.metadataLookup.instanceType = true
 		default:
-			t.Log.Errorf("ec2tagger: Unsupported EC2 Metadata key: %s", tag)
+			t.Log.Errorf(metadataCheckStrTagNotSupported, tag)
 		}
 	}
 
-	if !t.ec2metadata.Available() {
-		msg := "ec2tagger: Unable to retrieve InstanceId. This plugin must only be used on an EC2 instance"
-		t.Log.Errorf(msg)
-		return errors.New(msg)
-	}
+	t.Log.Infof(metadataCheckStrStartInitialization)
 
-	doc, err := t.ec2metadata.GetInstanceIdentityDocument()
-	if nil != err {
-		msg := fmt.Sprintf("ec2tagger: Unable to retrieve InstanceId : %+v", err.Error())
-		t.Log.Errorf(msg)
-		return errors.New(msg)
+	md := t.ec2MetadataProvider()
+	doc, err := md.GetInstanceIdentityDocument()
+	if err != nil {
+		t.Log.Errorf(metadataCheckStrInstanceDocumentFailure + metadataCheckStrEC2InstanceTagger)
```

Comment:
This is not readable to me. Just because the SSM agent code does it, doesn't mean that we have to follow it.

Reply:
Yes. I only want to introduce the changes and see everyone's comment. However, I have reduced the log to an [readable level](https://github.com/aws/amazon-cloudwatch-agent/pull/480/files#diff-6638b992018408bf44cac12361171fd050ff44fcc24c28f326988f4768a6dc2dR257-R261) which will show 
```
2022-06-07T03:36:11Z I! 
2022-06-07T03:36:11Z E! [processors.ec2tagger] ec2tagger: Unable to retrieve Instance Metadata Tags. This plugin must only be used on an EC2 instance.
2022-06-07T03:36:11Z E! [processors.ec2tagger] ec2tagger: Please increase hop limit to 3 by following this document https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-options.html#configuring-IMDS-existing-instances.
2022-06-07T03:36:11Z E! [telegraf] Error running agent: could not initialize processor ec2tagger: EC2MetadataRequestError: failed to get EC2 instance identity document
caused by: EC2MetadataError: failed to make EC2Metadata request
        status code: 401, request id: 
caused by:
```

Let's me know if it make sense.

Reply:
Please revert this change. Based on the log output you have shown, I can't tell how to derive the log message from the two variables that you are using in this log line. Why is there a `[telegraf]` level error log introduced?

Reply:
Yes I reverted the changes yesterday as can be seen here https://github.com/aws/amazon-cloudwatch-agent/pull/480/files#diff-6638b992018408bf44cac12361171fd050ff44fcc24c28f326988f4768a6dc2dR257-R261 (only 1 constant string).

Reply:
For the second question, its the default behavior when CloudWatchAgent fails to execute a daemon process that can be seen here https://github.com/aws/amazon-cloudwatch-agent/blob/master/cmd/amazon-cloudwatch-agent/amazon-cloudwatch-agent.go#L179-L182
```
err := runAgent(ctx, inputFilters, outputFilters)
if err != nil && err != context.Canceled {
			log.Fatalf("E! [telegraf] Error running agent: %v", err)
}
```